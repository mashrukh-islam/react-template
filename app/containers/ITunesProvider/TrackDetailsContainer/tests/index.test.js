import React from 'react';
import { timeout, renderProvider } from '@utils/testUtils';
import { TrackDetailsContainerTest as TrackDetailsContainer } from '../index';
import { cleanup } from '@testing-library/react';

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useParams: jest.fn(),
//   useRouteMatch: jest.fn()
// }));

describe('<TrackDetailsContainer /> tests', () => {
  let submitSpy;
  let trackDetails;

  afterEach(() => {
    cleanup();
  });

  beforeEach(() => {
    submitSpy = jest.fn();
    trackDetails = { artistViewUrl: '' };
  });

  it('should render and match snapshot', () => {
    const { baseElement } = renderProvider(
      <TrackDetailsContainer dispatchFetchTrackDetails={submitSpy} trackDetails={trackDetails} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchFetchTrackDetails on mounting', async () => {
    const getTrackDetailsSpy = jest.fn();
    const clearTrackDetailsSpy = jest.fn();
    renderProvider(
      <TrackDetailsContainer
        dispatchFetchTrackDetails={getTrackDetailsSpy}
        dispatchClearTrackDetails={clearTrackDetailsSpy}
        trackDetails={trackDetails}
      />
    );
    await timeout(1000);
    expect(getTrackDetailsSpy).toBeCalled();
  });
});
