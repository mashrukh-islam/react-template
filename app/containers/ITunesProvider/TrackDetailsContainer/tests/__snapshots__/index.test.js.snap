// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetailsContainer /> tests should render and match snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetailsContainer__Container-par8y9-0 beoNmx"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-cover"
        >
          <img />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="TrackDetailsContainer__Row-par8y9-1 fkOSsM"
          >
            <div
              class="TrackDetailsContainer__Col-par8y9-2 cFWTCU"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Track Name
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Album
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="TrackDetailsContainer__Col-par8y9-2 cFWTCU"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Track Time
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    NaN:NaN
                  </span>
                </div>
              </div>
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Genre
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="TrackDetailsContainer__Row-par8y9-1 fkOSsM"
          >
            <div
              class="TrackDetailsContainer__Col-par8y9-2 cFWTCU"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Explicitness
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
            <div
              class="TrackDetailsContainer__Col-par8y9-2 cFWTCU"
              size="1"
            >
              <div
                class="ant-statistic"
              >
                <div
                  class="ant-statistic-title"
                >
                  <p
                    class="T__StyledText-gjlic1-0 cbAvsY"
                    data-testid="t"
                  >
                    Artist
                  </p>
                </div>
                <div
                  class="ant-statistic-content"
                >
                  <span
                    class="ant-statistic-content-value"
                  >
                    <span
                      class="ant-statistic-content-value-int"
                    >
                      
                      0
                    </span>
                    
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          class="ant-card-actions"
        >
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a
                href=""
              >
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="t"
                >
                  View Artist
                </p>
              </a>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a>
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="t"
                >
                  View Album
                </p>
              </a>
            </span>
          </li>
          <li
            style="width: 33.333333333333336%;"
          >
            <span>
              <a>
                <p
                  class="T__StyledText-gjlic1-0 cbAvsY"
                  data-testid="t"
                >
                  View Track
                </p>
              </a>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
